lib/pbench/cli/server/report.py

```diff
@@ -779,8 +779,8 @@ def report(context: object, **kwargs):
         if kwargs.get("all") or kwargs.get("cache"):
             report_cache(cache_m)
         stats = kwargs.get("statistics")
-        if stats:
-            if stats == "creation":
+        if stats or kwargs.get("all"):
+            if stats == "creation" or stats is None:
                 report_creation(kwargs)
             elif stats == "upload":
                 report_uploads(kwargs)
```

Comment:
I think your intention is to display only one of creation stats or upload stats...so, if the user provides both `--all` and `--statistics=upload`, we presumably should not display both.  (That is, if we do want to display both in that case, we probably _also_ want to display both for `--all --statistics=creation` and maybe for just `--all` as well...so, either way, we should probably tweak the code.  😛)

I think something like this might be better:
```python
        stats = kwargs.get("statistics")
        if stats == "upload":
                report_uploads(kwargs)
        elif stats == "creation" or kwargs.get("all"):
                report_creation(kwargs)
```
That is:
- we don't need to bother with the nested check -- `get()` will return `None` if the option isn't specified, and that value is easy enough to work with;
- check for `"upload"` first, so that we can defer dealing with the default case;
- without the nested check, we can check for `get("all")` instead of also explicitly checking for `None`;
- so, if the user requests upload statistics, they don't get creation statistics, even if they add `--all`, and, if they don't provide `--statistics` but do provide `--all`, then they get creation statistics.

Also, I would omit the final `else` clause, since Click should ensure that it cannot be executed.  (That is, it is taking up space for no purpose, and, even if it would have been executed, omitting it doesn't really hurt anything -- we would just ignore the value (which Click should have objected to) and proceed normally.

Reply:
You seem to be suggesting there's a functional difference, but I don't see it. If you specify `--all --statistics=upload` the outer `if` will trigger (both `kwargs.get("all")` and `kwargs.get("statistics")` are truthy), and the inner `if` won't trigger (`stats` is neither `None` nor `creation`), so we'd hit the `elif` where `stats` does in fact equal `upload`.

Yeah, I could drop the extra indentation, and that's not without appeal, but I was inclined to spend about 5 seconds making this change, and now I've spent vastly longer just responding to the comment so I'm not really sure how I feel about that ...

And I'm not sure why I initially decided not to trust Click's choice validation. Maybe on the first statistics pass I didn't use it: but that's not new, or changed, so it's mostly out of scope here. 😆

Reply:
> You seem to be suggesting there's a functional difference, but I don't see it.

Good, 'cause I'm not seeing it now, either.  🥴  Sorry for the timesink.  🙁

Given that the bug is not there, I withdraw my other suggestions (but, you can still do them, if you want 😉).